### PR TITLE
rw test (cli): add `db-push` flag

### PR DESCRIPTION
**Pull request by pachoclo**
⚠️ This is my first PR 😬

---

- Adds `--db-push` flag to `rw test`, which calls Prisma's own [db push](https://www.prisma.io/docs/reference/api-reference/command-reference#db-push) before running the tests. 
- Default value is `true`.
- Pass `false` to this flag if you do not need to sync your test db or have one created automatically for you.
- This flag is ignored if the API side doesn't exist

To-do:
- [x] add new flag `--db-push` to cli's  test command
- [x] update docs (PR: https://github.com/redwoodjs/redwoodjs.com/pull/707)
- [x] add unit test 

---
Fixes: #1433

### CODE REVIEW

File: packages/cli/src/commands/test.js [hunk 1]
```diff
@@ -53,6 +53,11 @@ export const builder = (yargs) => {
       type: 'boolean',
       default: false,
     })
+    .option('disableDbGeneration', {
```
Comment (pachoclo) [inline packages/cli/src/commands/test.js:56]: @peterp , @thedavidprice  please let me know if the flag's name is ok

**Comment by dac09** [comment]
Hey @pachoclo thanks for this. Could I ask that you update the description with _when_ you're likely to use this flag? Something like 

> Use this flag to disable seeding your test database with scenarios. Useful when xxx

Also maybe a positive flag is better than a negative. e.g.. `--db-push` or `--seed-scenarios` with  the default value of false might be more appropriate. I'm also keen to keep the word "generate" out of it

**Comment by pachoclo** [comment]
Hi @dac09. Thanks for the feedback. 

- I agree that `--db-push` is more precise (that's what we are calling after all)
- I'll need a hand with the use case(s) in the description--not sure why Peter opened the original issue

- Regarding:
> with the default value of false might be more appropriate.

If we name the flag positive `--db-push` and setting it to false, we would be changing the current `rw test` behavior. Are you sure that's what we want? i.e. prisma db-push would be skipped by default after this change.

> Note that yargs will automatically generate the negatives for you i.e. --no-db-push or --no-seed-scrnarios

Already learning new things here. Thanks!

**Comment by dac09** [comment]
Thanks for the updates @pachoclo - for the tests, you might want to wait for my PR here https://github.com/redwoodjs/redwood/pull/2630 to be merged, and your new test can just follow that pattern :) 

Great first PR 🥳 and thank you so much for being part of the community and contributing

**Comment by thedavidprice** [comment]
Looping in @peterp Could you confirm this is the use case you originally envisioned in #1433?

Looking great @pachoclo 🤩

**Comment by peterp** [comment]
Howdy, this is exactly the functionality that I would like to disable - It would be great if there is a way to disable this via configuration, but that's a different issue.

**Comment by thedavidprice** [comment]
Hi @pachoclo we've merged #2630 You should now be able to move forward with the unit tests!

> It would be great if there is a way to disable this via configuration as well.
> ...My gut feels is that this should happen in api/jest.config.js

@peterp I definitely see the logic in this. However, I immediately wonder how we'd document it, how people would find out about it, etc. I think that keeping config standardized to redwood.toml is the way to go for now. We already have other CLI-based configs available that way.

@pachoclo you would _not_ necessarily need to add config option to this PR. You could move this part of the conversation to a new Issue for discussion. Then move this PR to merge and start on config as part 2 PR. Up to you!

File: packages/cli/src/commands/test.js [hunk 2]
```diff
@@ -48,6 +48,18 @@ export const builder = (yargs) => {
       type: 'boolean',
       default: false,
     })
+    .option('clearCache', {
```
Comment (dac09) [inline packages/cli/src/commands/test.js:51]: Think we don't need this option anymore :)

Comment (pachoclo) [reply packages/cli/src/commands/test.js:51]: oh woops! will remove

Comment (dac09) [reply packages/cli/src/commands/test.js]: Also think you need to add `db-push` to line 80, so we don't forward it to jest! 

Thanks again :)

**Comment by thedavidprice** [comment]
we had a CI bug in `main`. updating from main now to see if that resolves the E2E issue.

**Review by dac09 — APPROVED** [review]

**Pull request merged**